lib/Serialization/SerializeSIL.cpp

```diff
@@ -2550,6 +2615,24 @@ void SILSerializer::writeSILBlock(const SILModule *SILMod) {
     processSILFunctionWorklist();
   }
 
+  // Write out differentiability witnesses.
+  // Note: this must be done after visiting SIL functions above so that
```

Comment:
Note: diff. witness deserialization seems a bit overly complicated right now (auxiliary `DifferentiabilityWitnessesToEmit` data structure, etc) because the `differentiability_witness_function` instruction hasn't been upstreamed yet.

I decided to upstream the code as-is because it's the final desired version.